internal/server/storage/drivers/driver_lvm_volumes.go

```diff
@@ -458,11 +458,17 @@ func (d *lvm) SetVolumeQuota(vol Volume, size string, allowUnsafeResize bool, op
 			}
 
 			// Activate volume if needed.
-			_, err := d.activateVolume(vol)
+			activated, err := d.activateVolume(vol)
 			if err != nil {
 				return err
 			}
 
+			if !activated {
+				defer func() {
+					_, _ = d.activateVolume(vol)
```

Comment:
Long function so I'm probably missing some context, but it would seem like you would want to defer a *de*activate here if !activated?

Reply:
Oh, I see, th ecomment above activateVolume() says:

785 // activateVolume activates an LVM logical volume if not already present. Returns true if activated, false if not.

So it's true if activateVolume() itself managed to activate it, not if it was
activated beforehand.